packages/driver/src/cy/listeners.js

```diff
@@ -21,9 +21,16 @@ const removeAllListeners = () => {
 }
 
 const addListener = (win, event, cb) => {
-  events.push([win, event, cb])
+  const id = Math.random()
 
-  win.addEventListener(event, cb)
+  win.__cypressEventMapRPC = win.__cypressEventMapRPC || {}
+  win.__cypressEventMapRPC[id] = cb
+
+  const fn = new win.Function('evt', `window.__cypressEventMapRPC[${id}].call(this, evt)`)
+
+  events.push([win, event, fn])
+
+  win.addEventListener(event, fn)
```

Comment:
seems like this could be moved into a function and reused here and in `server.js` `addXhrHandler`, they are pretty similar

also, a comment explaining what this does would be helpful

Reply:
Probably could be, yeah. Also @brian-mann had an alternate idea for how we might solve this which won't require `new Function`